src/lib/flight_tasks/tasks/AutoLineSmoothVel/FlightTaskAutoLineSmoothVel.cpp

```diff
@@ -328,8 +329,20 @@ void FlightTaskAutoLineSmoothVel::_updateTrajConstraints()
 	_trajectory[2].setMaxJerk(_param_mpc_jerk_auto.get());
 
 	if (_velocity_setpoint(2) < 0.f) { // up
-		_trajectory[2].setMaxAccel(_param_mpc_acc_up_max.get());
-		_trajectory[2].setMaxVel(_param_mpc_z_vel_max_up.get());
+		float z_accel_constraint = _param_mpc_acc_up_max.get();
+		float z_vel_constraint = _param_mpc_z_vel_max_up.get();
+
+		// The constraints are broken because they are used as hard limits by the position controller, so put this here
+		// until the constraints don't do things like cause controller integrators to saturate. Once the controller
+		// doesn't use z speed constraints, this can go in AutoMapper::_prepareTakeoffSetpoints(). Accel limit is to
+		// emulate the motor ramp (also done in the controller) so that the controller can actually track the setpoint.
+		if (_type == WaypointType::takeoff &&  _dist_to_ground < _param_mpc_land_alt1.get()) {
+			z_vel_constraint = _param_mpc_tko_speed.get();
+			z_accel_constraint = math::min(z_accel_constraint, _param_mpc_tko_speed.get() / _param_mpc_tko_ramp_t.get());
```

Comment:
FYI @bresch This fixes the tracking while the controller is doing the velocity ramp with constraints.

Before:
![bokeh_plot(8)](https://user-images.githubusercontent.com/1746276/74542968-df3e6480-4f44-11ea-964d-07ca627fff6b.png)

After:
![bokeh_plot(9)](https://user-images.githubusercontent.com/1746276/74542997-e9606300-4f44-11ea-8992-7dba655ab8c5.png)